src/services/data-query.js

```diff
@@ -81,48 +112,115 @@ function QueryService(){
+    if (hasExternalLayersSelected) {
+      return this.handleRequest(this.getEmptyRequest(), query);
+    } else {
+      const layers = getMapLayersByFilter(layersFilterObject, condition);
+      const request = getQueryLayersPromisesByBBOX(layers, {
+        bbox,
+        feature_count,
+        filterConfig,
+        multilayers,
+      });
+      return this.handleRequest(request, query);
+    }
   };
-
   /**
    *
-   * @param map
-   * @param coordinates
+   * @param {{ coordinates: unknown, layerIds: unknown[], multilayers: boolean, query_point_tolerance: number, feature_count: number }}
+   * @param layerIds: <Array>
    * @param multilayers
    * @param feature_count
+   * 
    * @returns {Promise<unknown>}
    */
-  this.coordinates = async function({coordinates, layerIds=[], multilayers=false, query_point_tolerance=QUERY_POINT_TOLERANCE, feature_count}={}){
+  this.coordinates = async function({
+    coordinates,
+    layerIds              = [],                   // see: `QueryResultsService::addLayerFeaturesToResultsAction()`
+    multilayers           = false,
+    query_point_tolerance = QUERY_POINT_TOLERANCE,
+    feature_count
+  } = {}) {
+    const hasExternalLayersSelected = this.hasExternalLayerSelected({ type: "vector" });
+    /**
+     * @since v3.8
+     * @type {{external: {add: boolean, filter: {SELECTED: boolean}}, coordinates, type: string}}
+     */
+    const query = {
+      coordinates,
+      type: 'coordinates',
+      external: {
+        add: true,
+        filter: {
+          SELECTED: hasExternalLayersSelected
+        }
+      }
+    };
+
+    // Return an empty request if an external layer is selected
+    if (hasExternalLayersSelected && 0 === layerIds.length) {
+      return this.handleRequest(this.getEmptyRequest(), query);
+    }
+
     const layersFilterObject =  {
       QUERYABLE: true,
-      SELECTEDORALL: layerIds.length === 0,
+      SELECTEDORALL: (0 === layerIds.length),
       VISIBLE: true
     };
-    Array.isArray(layerIds) && layerIds.forEach(layerId => {
-      if (!layersFilterObject.IDS) layersFilterObject.IDS = [];
-      layersFilterObject.IDS.push(layerId);
-    });
+
+
+    if (Array.isArray(layerIds)) {
+      layerIds.forEach(id => {
+        if (!layersFilterObject.IDS) layersFilterObject.IDS = [];
+        layersFilterObject.IDS.push(id);
+      });
+    }
+
+    /**
+     * get layers to handle request
+     */
     const layers = getMapLayersByFilter(layersFilterObject);
-    const request = getQueryLayersPromisesByCoordinates(layers, {
-      multilayers,
-      feature_count,
-      query_point_tolerance,
-      coordinates
-    });
-    return this.handleRequest(request, {
-      coordinates,
-      type: 'coordinates',
-    });
+    /**
+     * @since v3.8 in case of project layer selected need to be set add external property to false to avoid add
+     * external layer to results
+     */
+    query.external.add = !(1 === layers.length && layers[0].isSelected());
```

Comment:
If I understand correctly, you mean like this?

```suggestion
    // set external property `add: false` in case
    // of selected layer in order to avoid querying
    // a temporary layer (external layer)
    if (1 === layers.length && layers[0].isSelected()) {
      query.external.add = false;
    }
```

Reply:
@Raruto right